python-threatexchange/threatexchange/exchanges/impl/techagainstterrorism_api.py

```diff
@@ -90,27 +99,17 @@ def get_client(self) -> api.TATHashListAPI:
     def fetch_iter(
         self,
         _supported_signal_types: t.Sequence[t.Type[SignalType]],
-        checkpoint: t.Optional[state.TFetchCheckpoint],
+        checkpoint: t.Optional[TATCheckpoint],
     ) -> t.Iterator[_TypedDelta]:
 
         client = self.get_client()
-        result = client.get_hash_list()
-
-        translated = (_get_delta_mapping(entry) for entry in result)
 
-        yield state.FetchDelta(
-            dict(t for t in translated if t[0][0]),
-            checkpoint=state.NoCheckpointing(),
-        )
-
-
-def _is_compatible_signal_type(record: t.Dict[str, str]) -> bool:
-    compatible_video_types = ["mov", "m4v", "mp4", "webm"]
-
-    if record["algorithm"] == "MD5":
-        return record["file_type"] in compatible_video_types
-
-    return record["algorithm"] == "PDQ"
+        for result in client.fetch_hashes_iter(checkpoint.checkpoint):
```

Comment:
blocking: [2/3] Then you are passing in the checkpoint string to fetch_hashes_iter